subprojects/platform-native/src/integTest/groovy/org/gradle/nativeplatform/LibraryDependenciesIntegrationTest.groovy

```diff
@@ -248,7 +248,7 @@ project(":lib") {
 """
 
         when:
-        executer.withArgument('--configure-on-demand')
+        executer.withArgument('--configure-on-demand').expectDeprecationWarning()
```

Comment:
Is it possible to put all `expectDeprecationWarning` in one place? Like inside `withArguments` methods or `run` method. Of course, this may be not feasible because of various complex situation, just a discussion.

Reply:
Unfortunately, the expectations is reset after every call to `run`. So I only have two options:

- add it to every `run` call
- override `run` for the whole class (as I did in the other test)

Unless of course I'm missing some feature in that test fixture. Maybe @big-guy has an idea?